### ISSUE TITLE
contributors entries are case sensitive

### ISSUE THREAD

**Issue by Simn** [HaxeFoundation/haxelib#59]
I submitted a library using this haxelib.json yesterday:

```
{
  "name": "hxparse",
  "url": "https://github.com/Simn/hxparse",
  "license": "BSD",
  "description": "This library provides tools for creating lexers and parsers in haxe.",
  "version": "1.0.1",
  "releasenote": "initial release",
  "contributors": ["simn"]
}
```

Today I wanted to push an update using the same haxelib.json (with a modified releasenote), but haxelib went on complaining "User 'simn' is not a developer of project 'hxparse'".

This is because my username for haxelib is uppercase Simn. It worked after changing contributors accordingly, but it should consistently work or fail for both initial and update submission.

**Comment by Simn** [comment]
*tumbleweed*

**Comment by nadako** [comment]
> The database doesn't allow registering simn if Simn exists

Is this really true? I started to look at this issue, and my local sqlite-based repo allowed to register both `nadako` and `Nadako`, what am I missing?

**Comment by nadako** [comment]
Oh, it's probably different for MySQL. Anyway that raises the question about how to do case insensitive queries with SPOD. I came up with:
```haxe
User.manager.unsafeObjects("SELECT * FROM User WHERE lower(name) = " + sys.db.Manager.quoteAny(name.toLowerCase()), true)
```
But this doesn't look good. Maybe we could add a `.toLowerCase` support in RecordMacros that generates `lower(field)` cc @ncannasse

**Comment by ncannasse** [comment]
RTFM :) http://dev.mysql.com/doc/refman/5.7/en/case-sensitivity.html
MySQL uses case insensitive matching by default, which I think is good for user names (every website you login is case insensitive I suppose)

**Comment by nadako** [comment]
Haxelib server [doesn't seem to check for existing user names](https://github.com/HaxeFoundation/haxelib/blob/a7460e8371fd43bb8c0f7504b5854e1611376e93/src/haxelib/server/Repo.hx#L104-L109) and the `name` field [is not marked as UNIQUE](https://github.com/HaxeFoundation/haxelib/blob/a7460e8371fd43bb8c0f7504b5854e1611376e93/src/haxelib/server/SiteDb.hx#L28-L36). So it's possible to register several users by the same name, which is just wrong. Could we use `name` field as a primary key and get rid of the `id`?

**Comment by nadako** [comment]
This strikes me as a big deal actually. I wonder how many people "successfully" registered duplicating user names.

**Comment by ncannasse** [comment]
@jasononeil could you look into that urgently ?

**Comment by nadako** [comment]
Note that I added the check in d679e23b7f0f2cdf44c0a1167e96af17e4e8dbc6

**Comment by nadako** [comment]
We need to check the production database for users with duplicate names (case-insensitive) and remove all but one, and fix references to that user in Project and Developer tables.

**Comment by Simn** [comment]
Totally unrelated question of course, but does our server make backups of the haxelib database?

**Comment by ibilon** [comment]
Might as well double that question with the database for the haxe.org blog.

**Comment by markknol** [comment]
Might as well triple that question with the content/files of haxelib.

**Comment by nadako** [comment]
library files should be on a CDN really (see #33)

**Comment by andyli** [comment]
As @ncannasse told me when I picked up the server last month, no, there is currently no backup mechanism in place.
I think we should hire a sys admin...

**Comment by nadako** [comment]
> I think we should hire a sys admin...

Indeed! A sysadmin probably could help with restoring data for #253 from http server logs :)

**Comment by ibilon** [comment]
That's not too hard to parse ;)

**Comment by markknol** [comment]
If this is not a joke it would be awesome.

**Comment by ibilon** [comment]
Thinking more about this, with rewriting I don't think the logs have enough information for that.

**Comment by nadako** [comment]
Maybe we should just use some to AWS or something.

**Comment by ncannasse** [comment]
Yes, server management is an area we're lacking, because there's not that much things that needs to be done, so we need someone to help occasionally. It's not in that much bad shape, but there things that should be done. Any volunteer ? :)
